File: LeetCode/1071. Greatest Common Divisor of Strings.cpp
class Solution {
public:
    int gcd(int x, int y){
      if(x>y) swap(x, y);
      if(y%x==0) return x;
      return gcd(y%x, x);
    }
    string gcdOfStrings(string str1, string str2) {
        if(str1+str2 != str2+str1) return "";
        int n = str1.size(), m = str2.size();
        return str1.substr(0,gcd(n,m));
    }
};

File: LeetCode/1071. Greatest Common Divisor of Strings.py
class Solution:
    def gcd(x, y):
        if x>y: x, y = y, x
        if y%x==0: 
            return x
        return gcd(y%x, x)
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if(str1+str2 != str2+str1):
            return ""
        n = len(str1)
        m = len(str2)
        return str1[:gcd(n,m)]